### ISSUE TITLE
ECS TaskDefinition container definition cannot be updated

### ISSUE THREAD

**Issue by pierskarsenbarg** [pulumi/pulumi-aws-native#716]
### What happened?

When updating the container definitions input of an ECS Task Definition, getting an error message saying that the properties cannot be updated.

### Steps to reproduce

1. Clone repo: https://github.com/pierskarsenbarg/task-definition-update-fails
2. Run `pulumi up`
3. Update `index.js` in the `app` folder (which will cause Pulumi to push a new version to the repo and update the image URL in the container definition)
4. Run `pulumi up`

### Expected Behavior

A new version of the task definition will be created

### Actual Behavior

Error message:

```zsh
error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: c9d2c4e0-359b-4735-b54b-14d4e5a72ceb, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/ContainerDefinitions] cannot be updated
```

### Output of `pulumi about`

```
CLI          
Version      3.48.0
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.21.1
aws-native  0.42.0
awsx        1.0.0
docker      3.6.1
nodejs      unknown

Host     
OS       darwin
Version  13.0.1
Arch     x86_64

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/pierskarsenbarg
User           pierskarsenbarg
Organizations  pierskarsenbarg, karsenbarg, team-ce, demo

Pulumi locates its logs in /var/folders/69/3w1gr05s2pq36wn49bhyknym0000gn/T/ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by pierskarsenbarg** [comment]
Can confirm that @danielrbradley 's suggestion here works: https://github.com/pulumi/pulumi-aws-native/issues/628#issuecomment-1253514724 but a fix would be better

**Comment by mikhailshilkov** [comment]
Depends on implementing custom diffing in this provider tracked in https://github.com/pulumi/pulumi-aws-native/issues/736

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to `ContainerDefinitions` will result in a replace rather than an update.
